src/main/java/hypersquare/hypersquare/dev/code/player/action/PlayerSetMovementSpeed.java

```diff
@@ -0,0 +1,115 @@
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.util.Vector;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class PlayerSetMovementSpeed implements Action {
+    @Override
+    public void execute(@NotNull ExecutionContext ctx, @NotNull CodeSelection targetSel) {
+        for (Player p : targetSel.players()) {
+            float speed = ctx.args().<DecimalNumber>allNonNull("speed").getFirst().toFloat();
+            float groundSpeed = speed/500;
+            if(groundSpeed<0) groundSpeed = 0;
+            if(groundSpeed>1) groundSpeed = 1;
+            float flightSpeed = speed/1000;
+            if(flightSpeed<0) flightSpeed = 0;
+            if(flightSpeed>1) flightSpeed = 1;
+
+            PlayerSetMovementSpeed.SpeedTypes speedType = ctx.getTag("speed_type", PlayerSetMovementSpeed. SpeedTypes::valueOf);
+                if(speedType == SpeedTypes.GROUND) p.setWalkSpeed(groundSpeed);
+                if(speedType == SpeedTypes.FLIGHT) p.setFlySpeed(flightSpeed);
+                if(speedType == SpeedTypes.BOTH) {
+                    p.setWalkSpeed(groundSpeed);
+                    p.setFlySpeed(flightSpeed);
+            }
+        }
+    }
+
+    @Override
+    public BarrelParameter[] parameters() {
+        return new BarrelParameter[]{
+            new BarrelParameter(DisplayValue.NUMBER, false, false, Component.text("Movement speed percentage (0% to 1000%)"), "speed")
+        };
+    }
+
+    @Override
+    public BarrelTag[] tags() {
+        return new BarrelTag[]{
+            new BarrelTag("speed_type", "Speed Type", SpeedTypes.GROUND,
+                new BarrelTag.Option(SpeedTypes.GROUND, "Ground speed", Material.IRON_BOOTS),
+                new BarrelTag.Option(SpeedTypes.FLIGHT, "Flight speed", Material.ELYTRA),
+                new BarrelTag.Option(SpeedTypes.BOTH, "Both", Material.SUGAR)
+            )
+        };
+    }
+
+    @Override
+    public String getId() {
+        return "set_movement_speed";
+    }
+
+    @Override
+    public String getCodeblockId() {
+        return "player_action";
+    }
+
+    @Override
+    public String getSignName() {
+        return "SetSpeed";
+    }
+
+    @Override
+    public String getName() {
+        return "Set Movement Speed";
+    }
+
+    @Override
+    public ActionMenuItem getCategory() {
+        return PlayerActionItems.STATISTICS_CATEGORY;
+    }
+
+    @Override
+    public ItemStack item() {
+        return new ActionItem()
+            .setMaterial(Material.NETHERITE_BOOTS)
+            .setName(Component.text("Set Movement Speed").color(NamedTextColor.GOLD))
+            .setDescription(Component.text("Sets a player's walking"),
+                Component.text("and/or flight speed."))
+            .setParameters(parameters())
+            .setTagAmount(tags().length)
+            .build();
```

Comment:
Let's have an additional info saying how fly speed is 0%->1000% and walk speed is 0%->500%